### DISCUSSION TITLE
Metadata info not available to templates

### DISCUSSION THREAD

**Discussion by phisco** [aquasecurity/trivy#4218] [Ideas]
Currently, outputting to json format returns an object which also contains a few fields, for example the `Metadata` field which contains useful information about the package itself, such as `EOSL` and the artifact's SHA, unfortunately these info are not available for templates as they only get the `Results` list as input.

It would be awesome to have all information also available to templates directly.

## Example

Running `trivy image -f json <IMAGE>`, outputs:
```json
{
  "SchemaVersion": 2,
  "ArtifactName": "...",
  "ArtifactType": "container_image",
  "Metadata": {
    "OS": {
      "Family": "debian",
      "Name": "9.13",
      "EOSL": true
    },
    "ImageID": "sha256:...",
    "DiffIDs": [
      ...
    ],
    "RepoTags": [
      "..."
    ],
    "RepoDigests": [
      "...@sha256:..."
    ],
    "ImageConfig": {
      "architecture": "amd64",
      "author": "Bazel",
      "created": "2022-09-14T13:38:48.225647985Z",
      "history": [
        {
          "author": "Bazel",
          "created": "1970-01-01T00:00:00Z",
          "created_by": "bazel build ..."
        },
        ...
      ],
      "os": "linux",
      "rootfs": {
        "type": "layers",
        "diff_ids": [
           "..."
        ]
      },
      "config": {
        "Entrypoint": [
          "...."
        ],
        "Env": [
          "..."
        ],
        "User": "65532",
        "WorkingDir": "/",
        "ExposedPorts": {
          "8080/tcp": {}
        }
      }
    }
  },
  "Results": [
    ...  List of Results ...
  ]
}
```

While running `trivy image -f template --template '@<PATH_TO_TEMPLATE>' <IMAGE>`, pointing to the following template:
```
{{ toPrettyJson . }}
```
Outputs only the content of `.Results` above:
```json
[
  ... List of Results as above ...
]

```

**Comment by knqyf263** [comment]
It sounds reasonable, but it will be a breaking change. We need to think about migration.

**Comment by phisco** [comment]
Maybe something like `-f templateFull` or a different flag to pass the template file `--template-full "@<FILEPATH>`?

**Comment by phisco** [comment]
I can open a PR for that if we agree on an approach 🙏

**Reply by knqyf263** [reply]
I don't want to add more CLI flags, but we may want to add `template_full=true` in [the structured output options](https://github.com/aquasecurity/trivy/issues/3243).
